crates/router/src/core/routing/helpers.rs

```diff
@@ -170,7 +175,19 @@ pub async fn update_merchant_active_algorithm_ref(
 
     Ok(())
 }
-// TODO: Move it to business_profile
+
+#[cfg(all(any(feature = "v1", feature = "v2"), feature = "merchant_account_v2"))]
+#[cfg(all(feature = "v2", feature = "merchant_account_v2"))]
+pub async fn update_merchant_active_algorithm_ref(
```

Comment:
We would need to remove this function and update all the routing_algorithm's, i.e., surcharge and 3ds Decision Manager, on business_profile and not merchant_account. This can be taken up later.